Arduino_BHY2/src/Arduino_BHY2.h

```diff
@@ -40,35 +55,137 @@ class NiclaSettings {
   uint8_t conf = 0;
 };
 
+/**
+ * @brief Main class for initialising communication with sensors
+*/
 class Arduino_BHY2 {
 public:
   Arduino_BHY2();
   virtual ~Arduino_BHY2();
 
   // Necessary API. Update function should be continuously polled 
+  /**
+   * @param config Configuration for set @ref NiclaConfig state
+   * @param niclaConnection Configuration for set @ref NiclaWiring state
+   * 
+  */
   bool begin(NiclaConfig config = NICLA_BLE_AND_I2C, NiclaWiring niclaConnection = NICLA_VIA_ESLOV);
   bool begin(NiclaSettings& settings);
+  /**
+   *  @brief  Pings sensor
+   * 
+   *  @note Sensor must be continuously polled to prevent sleep
+   */
   void update(); // remove this to enforce a sleep
+  /**
+   *  @brief  Pings sensor and then sleep
+   * 
+   *  @note Delay does not stall microcontroller to sleep. See https://docs.arduino.cc/built-in-examples/digital/BlinkWithoutDelay
+   */
   void update(unsigned long ms); // Update and then sleep
+  /**
+  *   @brief delay method to be used instead of Arduino delay().
+  */
   void delay(unsigned long ms); // to be used instead of arduino delay()
 
   // API for using the bosch sensortec from sketch
+  /**
+   * @brief Poll Bosch method 
+   * 
+   * @param config Sensor configuration
+   */
   void configureSensor(SensorConfigurationPacket& config);
+  /**
+   * @brief Setup virtual sensors on the BHI260
+   * 
+   * @param sensorId  Sensor ID for sensor
+   * @param sampleRate Polling rate for sensor
+   * @param latency   Latency in milliseconds
+   * 
+   * @note For list of SensorID, see src/SensorID.h
+   */
   void configureSensor(uint8_t sensorId, float sampleRate, uint32_t latency);
+  /**
+   * @brief Handlo FIFO of data queue
+   * 
+   * @param sensorData Data packet from sensor
+   */
   void addSensorData(SensorDataPacket &sensorData);
+  /**
+   * @brief Handlo FIFO of data queue for long sensor data
+   * 
+   * @param sensorData Data packet from sensor
+   */
   void addLongSensorData(SensorLongDataPacket &sensorData);
+  /**
+   * @brief Return available sensor data
+   * 
+   * @return uint8_t 
+   */
   uint8_t availableSensorData();
+  /**
+   * @brief Return available long sensor data
+   * 
+   * @return uint8_t 
+   */
   uint8_t availableLongSensorData();
+  /**
+   * @brief Read sensor data
+   * 
+   * @param data 
+   * @return true 
+   * @return false 
+   */
   bool readSensorData(SensorDataPacket &data);
+  /**
+   * @brief Read long sensor data
+   * 
+   * @param data 
+   * @return true 
+   * @return false 
+   */
   bool readLongSensorData(SensorLongDataPacket &data);
+  /**
+   * @brief Check existance of sensor
+   * 
+   * @param sensorId Selected virtual sensor
+   * @return true - Sensor is present
+   * @return false - Sensor is not present
+   */
   bool hasSensor(uint8_t sensorId);
-
+  /**
+   * @brief Parse XYZ cartesian data
```

Comment:
```suggestion
   * @brief Parse XYZ Cartesian data
```

"Cartesian" is capitalized because it is derived from a proper noun (René Descartes).

References:

- https://style.mla.org/scientific-laws-theories-and-terms/
- https://en.wikipedia.org/wiki/Cartesian_coordinate_system